src/engine/Settings.cs

```diff
@@ -337,6 +348,25 @@ public void ApplySoundSettings()
         AudioServer.SetBusMute(gui, VolumeGUIMuted);
     }
 
+    /// <summary>
+    ///   Applies the current control scheme to the InputMap.
+    /// </summary>
+    public void ApplyInputSettings()
+    {
+        var data = InputGroupList.Instance.GetCurrentlyPendingControlScheme();
+        foreach (var action in data)
+        {
+            // Clear all old input events
+            InputMap.ActionEraseEvents(action.Key);
+
+            // Add the new input events
+            foreach (var inputEvent in action.Value)
+            {
+                InputMap.ActionAddEvent(action.Key, inputEvent);
```

Comment:
Does this work if there is an action specified in the input json that doesn't exist in Godot?
I think it might make sense to remove the inputs configured in godot editor, as the json needs to be also updated, right?

Reply:
no!
The godot inputs is where the game gets the default inputs from

Reply:
I see. I don't remember seeing duplicate input definitions. I made this comment before reading and digesting all of the changes in this PR.
It's good to keep things as is (default keys configured just once in Godot editor).

Reply:
You can still just define new inputs in the godot editor and everything works, but you cannot rebind them in the options.
If you want the user to be able to rebind the new input you need to insert it to the json

Reply:
That's exactly how it should work.
Though, I have one more concern. Does the system correctly handle new inputs being added later? That should work so that the new inputs are added to the active inputs even if the player has edited other inputs.
So the logic would be: on startup add any input keys that don't exist in the current input scheme to the scheme to ensure that things work fine when new actions are added.

Reply:
I think runtime additions to the InputMap should work fine.
I can't think of any reason why runtime additions to the InputMap should fail.
Ill test that later

Reply:
I'm not talking about runtime additions. I'm talking about a potential bug:
- Player plays the game
- Edits the keys a bit
- A new release is released, it adds a new input action name to the game
- The user starts the game, and now they can't use that new input or the input binding doesn't show it

To work against this, on start the game should go through the InputMap and add any inputs to the current control scheme that are missing.

Reply:
I have tested it and the removal and addition of inputs works

Reply:
That's good. It's really difficult to see from the quite large amount of code if that is handled correctly or not.